lib/widget/dialogs/cannot_open_file_dialog.dart

```diff
@@ -3,20 +3,25 @@ import 'package:localsend_app/gen/strings.g.dart';
 import 'package:localsend_app/util/platform_check.dart';
 import 'package:localsend_app/widget/dialogs/custom_bottom_sheet.dart';
 import 'package:routerino/routerino.dart';
+import '../../provider/receive_history_provider.dart';
 
 class CannotOpenFileDialog extends StatelessWidget {
   final String path;
 
   const CannotOpenFileDialog({required this.path, super.key});
 
-  static void open(BuildContext context, String path) {
+  static void open(BuildContext context, String path, String? fileId, ReceiveHistoryNotifier? filesRef) {
     if (checkPlatformIsDesktop()) {
       showDialog(
         context: context,
         builder: (_) => AlertDialog(
           title: Text(t.dialogs.cannotOpenFile.title),
           content: Text(t.dialogs.cannotOpenFile.content(file: path)),
           actions: [
+            TextButton(
```

Comment:
Since removing the file from history is optional, you can make this button optional too.

```dart
actions: [
  if (fileId != null)
    TextButton(
       // ...
     ),
]
```

Reply:
But would fileId ever be null? The dialogue can only appear if you click a file entry in the history, so there will always be a file entry to be deleted with this action, or am I mistaken?

Reply:
As a sender, you can open the file before sending the file to the recipient (see `selected_files_page.dart`).

Reply:
> As a sender, you can open the file before sending the file to the recipient (see `selected_files_page.dart`).

I see, I previously asserted the values as non-null thinking otherwise.